core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMergeSink.java

```diff
@@ -21,9 +21,27 @@
 
 public interface ConnectorMergeSink
 {
+    /**
+     * Represents INSERT query or INSERT via MERGE
```

Comment:
It's per-row, not per-query, so

```suggestion
     * Represents inserted row (for INSERT or MERGE query)
```